primitives/src/lib.rs

```diff
@@ -80,9 +88,31 @@ impl Default for CommitteeSeats {
     }
 }
 
+/// Configurable parameters for kick-out validator mechanism
+#[derive(Decode, Encode, TypeInfo, Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct CommitteeKickOutThresholds {
+    /// in a session, what is a limit for produced blocks under which the session counts as bad uptime
+    pub block_count_threshold: BlockCount,
+    /// how many bad uptime sessions force validator to be removed from the committee
+    pub underperformed_session_count_threshold: SessionCount,
```

Comment:
I know this is hard technically, but we should think how to do the counting "per period of time" and not "for ever". If the threshold is say `3` (a very reasonable number btw.), then a validator which missed 3 sessions in 2 years of time, will be kicked the same as if it would miss 3 sessions in one day. I don't think it's desired.

One solution I see is that we could have a sudo extrinsic `reset_underperformed_session_counts` that just zeros it for everyone, and maybe reset the thing every month or so. That's a bad solution though, looking for better ones :)

Reply:
Yes! That's what I meant about "not resetting"...

> That's a bad solution though, looking for better ones :)

Why not fix to `3` like you said. If Substrate runtime updates are as good as people say, that value could get changed with a `sudo`/update later on?

Reply:
On the other hand, having 0 as the block limit makes not harsh at all in my opinion, and this is because making 0 blocks in a session really means in practice two situations A) keys are misconfigured or B) the validator process is down. In both cases, it's on the validator to fix those, so having such 3 sessions, which in practice in the current committee size means a whole era, it's not big deal IMHO. Agree that setting to a non-zero block threshold can be harsh since it implies minor network issues which could make the validator to miss some slots and hence count the session as bad performance. 

My proposal is then: for the initial version, which was tested already, let's make block count hardcoded to 0 (ie, not configurable), and keep session count configurable as it is now and write 3 here as the default. Also for the initial version, don't add any resetting `underperformed_session_count_threshold` for now since we need to figure out a better way to do this, and we don't want to block this PR on this. If some validator would be kicked out by 3 sessions, which I doubt it would be **not** due to the validator's fault, we can always add it back in the next era.

Reply:
> Fix what to 3 exactly :P?

Width of the sliding window over the last 	`N` (=3) sessions which we would take into consideration when deciding whether validator underperformed. i.e. it would reset automatically after `N`.

Reply:
Yes. This automatic reset is complicated because you need to add items to storage for that. The more storage you add, the less efficient it is, and also the more complex and the harder to change in the future. We found a workaround to make it stateless by resetting if session number `=0 (mod N)` -- this has some disadvantages, but keeps it simple and does not require additional storage.

Reply:
done